server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueManagement.java

```diff
@@ -56,4 +56,13 @@ public void deleteAllQueues() {
         api.listQueues()
             .forEach(queue -> api.deleteQueue("/", queue.getName()));
     }
+
+    public Long getQueueSize(MailQueueName name) {
+        return api.listQueues()
```

Comment:
Instead of listing all the queues and filtering the one we need could we just call the api just for the queue we are interested in :
instead of   calling /api/queues we may call /api/queues/vhost/name

Reply:
I've tried it, sadly we're on '/' vhost and I failed to query it.

Reply:
I guess you can query it with %3F

Reply:
have you tried  ?
https://stackoverflow.com/questions/33119611/how-to-make-rabbitmq-api-calls-with-vhost#answer-33119612

Reply:
Do the deleteQueue of the api method which pass a vhost works ?

Reply:
https://github.com/rabbitmq/rabbitmq-management/issues/699